frontend/src/pages/modelServing/screens/global/useRouteForInferenceService.ts

```diff
@@ -14,34 +11,40 @@ const useRouteForInferenceService = (
   const [loaded, setLoaded] = React.useState(false);
   const [loadError, setLoadError] = React.useState<Error | null>(null);
 
-  const routeName = inferenceService.metadata.name;
+  const routeName = isKServe
+    ? `${inferenceService.metadata.name}-predictor`
+    : inferenceService.metadata.name;
   const routeNamespace = inferenceService.metadata.namespace;
-  const kserveRoute = isKServe ? getUrlFromKserveInferenceService(inferenceService) : null;
-  const state = getInferenceServiceActiveModelState(inferenceService);
-  const kserveLoaded = state === InferenceServiceModelState.LOADED;
 
   React.useEffect(() => {
     if (isKServe) {
-      setRoute(kserveRoute || null);
-      setLoaded(kserveLoaded);
-      setLoadError(kserveLoaded ? (kserveRoute ? null : new Error('Route not found')) : null);
-      return;
+      getKNativeService(routeName, routeNamespace)
+        .then((route) => {
+          setRoute(route.status.url);
+          setLoaded(true);
+        })
+        .catch((e) => {
+          setLoadError(e);
+          setLoaded(true);
+        });
     }
+
     if (!isRouteEnabled) {
```

Comment:
Should we bring this check up to the top of the `useEffect` hook? only when the route is enabled then we want to do the network calls, right?

Reply:
no, right now we don't have route enablement for kserve, so that only works for modelmesh, it should be there